pkg/controller/operators/catalog/operator.go

```diff
@@ -1694,9 +1689,7 @@ func (o *Operator) removeSubsCond(subs []*v1alpha1.Subscription, condType v1alph
 		}
 		sub.Status.LastUpdated = lastUpdated
 		sub.Status.RemoveConditions(condType)
-		subList = append(subList, sub)
 	}
-	return subList
```

Comment:
Since #2942 we no longer use the return value of `removeSubsCond` - how removing return value helps to fix #3162?

I think I'm missing something. I read the explanation in the PR description, but I still do not understand the connection. Could you please explain it to me in more details?

Reply:
@m1kola so it's definitely a little bit confusing to me too. The thing that actually fixed the problem isn't just "removing the return value" though. It's the fact that `setSubsCond` is not being used here and instead has been switched over to `removeSubsCond`. From what I can tell by trying to keep track of everything going on with original pointers being modified and new list being returned, it looked like there were two lists we were dealing with [here](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/operators/catalog/operator.go#L1396-L1405), and one of those lists were actually reaching [`UpdateSubscriptionStatues`](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/operators/catalog/operator.go#L1424), leading to the `removeSubsCond(resolutionError)` to fly under the radar without making any dent. 

So the thing that actually fixed the problem, is getting rid of the confusions by using the list of pointers like it's supposed to be used, which it seems was the intention in #2942 too.

Reply:
Thanks for the explanation, I did not take the fact that `setSubsCond` is being used into account.

Resolving this thread.